cmd/libsnap-confine-private/device-cgroup-support.c

```diff
@@ -58,26 +61,27 @@ struct sc_device_cgroup {
 
 __attribute__((format(printf, 2, 3))) static void sc_dprintf(int fd, const char *format, ...);
 
-static sc_cgroup_fds sc_udev_open_cgroup_v1(const char *security_tag);
+static int sc_udev_open_cgroup_v1(const char *security_tag, int flags, sc_cgroup_fds *fds);
 static void sc_cleanup_cgroup_fds(sc_cgroup_fds *fds);
 
-static void _sc_cgroup_v1_init(sc_device_cgroup *self) {
+static int _sc_cgroup_v1_init(sc_device_cgroup *self, int flags) {
     self->v1.fds = sc_cgroup_fds_new();
 
     /* initialize to something sane */
-    sc_cgroup_fds fds = sc_udev_open_cgroup_v1(self->security_tag);
-    if (fds.cgroup_procs_fd < 0) {
+    if (sc_udev_open_cgroup_v1(self->security_tag, flags, &self->v1.fds) < 0) {
+        if (flags == SC_DEVICE_CGROUP_FROM_EXISTING) {
```

Comment:
My remaining question would be to explain this with a comment: why isn't this a fatal error?